### ISSUE TITLE
K/N artifact klib file doesn't exists on version 5.5.5

### ISSUE THREAD

**Issue by RyuNen344** [kotest/kotest#3388]
I bumped to 5.5.5, gradle occurred error.

```
   > Could not find kotest-common.klib (io.kotest:kotest-common-iosx64:5.5.5).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/io/kotest/kotest-common-iosx64/5.5.5/kotest-common-iosx64-5.5.5.klib


Caused by: org.gradle.internal.resolve.ArtifactNotFoundException: Could not find kotest-common.klib (io.kotest:kotest-common-iosx64:5.5.5).
```

Could you check artifact?

**Comment by nicolasfara** [comment]
I have a similar problem related to `macosArm64`

```
* What went wrong:
Execution failed for task ':core:compileTestKotlinMacosArm64'.
> Could not resolve all files for configuration ':core:macosArm64TestCompileKlibraries'.
   > Could not find kotest-common.klib (io.kotest:kotest-common-macosarm64:5.5.5).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/io/kotest/kotest-common-macosarm64/5.5.5/kotest-common-macosarm64-5.5.5.klib
 ```

**Comment by Kantis** [comment]
I dont think anything has really changed regarding the publishing of these. I see they exist in 5.5.4. My guess is something went wrong during the manual steps of the release. Nothing we can do to fix it now unfortunately, since the artifacts are published already, right @sksamuel ?

**Comment by nicolasfara** [comment]
One solution could be bump to 5.5.6 and be sure tu upload all the artifacts.

This problem has broken my CI pipeline preventing me to update to 5.5.5

Moreover, I have noticed that a similar problem was encountered in Koin (https://github.com/InsertKoinIO/koin/issues/1497), for this reason I suggest to evaluate the use of this plugin: https://github.com/DanySK/publish-on-central to automate the release process of the artifacts to the Maven central. I'm using that plugin for a Kotlin multiplatform project (https://github.com/nicolasfara/pulverization-framework) and I have never experienced problems like this one with that plugin :)

**Comment by LouisCAD** [comment]
FYI, in Splitties, another multiplatform and multi-modules library, I did steps to prevent this from happening. Here's how my publishing process looks like: https://github.com/LouisCAD/Splitties/actions/runs/1151968403

I can help you set up something similar for kotest if you want.

**Comment by LouisCAD** [comment]
Have you seen my proposal, @Kantis?

**Comment by sksamuel** [comment]
That looks great. What does `./gradlew --scan :tools:publication-checker:build` do ?

**Comment by LouisCAD** [comment]
It checks the build against the staging repo for each supported platform. For kotest 5.5.5, it would fail. That said, I'm not sure it's necessary as I've never seen it fail with the way to publication is done. I'd ditch that step, and add it only if we ever have a failure later.

**Comment by JavierSegoviaCordoba** [comment]
I haven't checked it via Gradle but looks like klib is missing on 5.6.0 here:

https://repo1.maven.org/maven2/io/kotest/kotest-common-iosx64/5.6.0/

**Comment by RyuNen344** [comment]
```log
   > Could not find kotest-common.klib (io.kotest:kotest-common-iosx64:5.6.0).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/io/kotest/kotest-common-iosx64/5.6.0/kotest-common-iosx64-5.6.0.klib
```

same error in 5.6.0 ;-<

**Comment by sksamuel** [comment]
Yeah sonatype is not the most reliable of hosts. I am republishing as 5.6.1 now and doing it individually to ensure all the parts are there.

**Comment by sksamuel** [comment]
5.6.1 should be good for everyone !

**Comment by sksamuel** [comment]
Remember that Kotest 5.6 requires Kotlin 1.8+

**Comment by MSDarwish2000** [comment]
@sksamuel Sorry to say that but there seems to be another problem with publishing artifacts that affected 5.6.1. When publishing a library that uses Kotest Assertions I've got this exception

```
Execution failed for task ':kache:transformCommonMainDependenciesMetadata'.
> Could not resolve all files for configuration ':kache:tvosArm64TestCompilationDependenciesMetadata'.
   > Could not find io.kotest:kotest-assertions-core-tvosarm64:5.6.1.
     Required by:
         project :kache > io.kotest:kotest-assertions-core:5.6.1
```

It can be found in [this GitHub workflow run](https://github.com/MayakaApps/Kache/actions/runs/4736276860/jobs/8407594302). Actually, I'm not sure why it appeared while publishing but not when running tests.as kotest asserions is a dependency of commonTest as expected not commonMain.

Edit: I've opened issue #3493 for this problem.
